core-plugins/src/main/java/io/cdap/plugin/transform/LogParserTransform.java

```diff
@@ -216,12 +228,14 @@ private String getLog(StructuredRecord input) {
     }
   }
 
-  private void validateInputSchemaType(Schema.Type inputSchemaType) {
+  private void validateInputSchemaType(Schema inputSchema, String inputName, @Nonnull FailureCollector collector) {
+    Schema.Type inputSchemaType = inputSchema.getType();
     if (!Schema.Type.STRING.equals(inputSchemaType) && !Schema.Type.BYTES.equals(inputSchemaType)) {
-      throw new IllegalArgumentException(String.format(
-        "Unsupported inputType in schema, only Schema.Type.BYTES and Schema.Type.STRING are supported " +
-          "InputType: %s", inputSchemaType.toString()));
-
+      collector.addFailure(
+          String.format("Field '%s' of unsupported type '%s'.", inputName, inputSchema.getDisplayName()),
+          String.format("Ensure it is of type %s or %s.", Schema.Type.BYTES, Schema.Type.STRING))
```

Comment:
use displayname method to keep the error messages consistent

Reply:
I believe that it is modified to use getDisplayName(). Are you referring to the enum types in the correctiveAction field?